File: src/main/java/no/novari/flyt/history/mapping/selectable/InstanceStatusSelectableMappingService.java
package no.novari.flyt.history.mapping.selectable;

import no.novari.flyt.history.model.Selectable;
import no.novari.flyt.history.model.instance.InstanceStatus;
import org.springframework.stereotype.Service;

@Service
public class InstanceStatusSelectableMappingService {

    public Selectable<String> toSelectable(InstanceStatus instanceStatus) {
        return Selectable
                .<String>builder()
                .value(instanceStatus.name())
                .label(getDisplayText(instanceStatus))
                .build();
    }

    public String getDisplayText(InstanceStatus instanceStatus) {
        return switch (instanceStatus) {
            case IN_PROGRESS -> "Under behandling";
            case TRANSFERRED -> "Overført";
            case ABORTED -> "Avbrutt";
            case FAILED -> "Feilet";
        };
    }

}

File: src/main/kotlin/no/novari/flyt/history/mapping/selectable/InstanceStatusSelectableMappingService.kt
package no.novari.flyt.history.mapping.selectable

import no.novari.flyt.history.model.Selectable
import no.novari.flyt.history.model.instance.InstanceStatus
import org.springframework.stereotype.Service

@Service
class InstanceStatusSelectableMappingService {
    fun toSelectable(instanceStatus: InstanceStatus): Selectable<String> {
        return Selectable(
            value = instanceStatus.name,
            label = getDisplayText(instanceStatus),
        )
    }

    fun getDisplayText(instanceStatus: InstanceStatus): String {
        return when (instanceStatus) {
            InstanceStatus.IN_PROGRESS -> "Under behandling"
            InstanceStatus.TRANSFERRED -> "Overført"
            InstanceStatus.ABORTED -> "Avbrutt"
            InstanceStatus.FAILED -> "Feilet"
        }
    }
}
